applets/wncklet/wayland-backend.c

```diff
@@ -377,8 +383,86 @@ foreign_toplevel_handle_closed (void *data,
 				struct zwlr_foreign_toplevel_handle_v1 *toplevel)
 {
 	ToplevelTask *task = data;
+
 	if (task->button)
+	{
+		GtkOrientation orient;
+		GtkWidget *button;
+		GtkWidget  *box;
+		GtkWidget *outer_box = gtk_widget_get_parent(GTK_WIDGET(task->button));
 		gtk_widget_destroy (task->button);
+		buttons = buttons -1;
+
+		if (buttons == 0)
+			return;
+
+		/* We don't need to modify button size on a vertical panel*/
+		orient = gtk_orientable_get_orientation(GTK_ORIENTABLE(outer_box));
+		if (orient == GTK_ORIENTATION_VERTICAL)
+			return;
+
+		/* Horizontal panel: if buttons can now fit
+		 * with both labels and icons show them
+		 */
+		if (tasklist_width / buttons > icon_size * 3)
+		{
+			GList* children = gtk_container_get_children (GTK_CONTAINER(outer_box));
+			while (children != NULL)
+			{
+				button = GTK_WIDGET (children->data);
+
+				/* If maximum width buttons fix, expand to that dimension*/
+				if (buttons * max_button_width < tasklist_width)
+					gtk_widget_set_size_request (button, max_button_width, -1);
+
+				/* Otherwise expand remaining buttons to fill the tasklist*/
+				else
+					gtk_widget_set_size_request (button, tasklist_width / buttons, -1);
+
+				gtk_widget_show_all (button);
+				children = children->next;
+			}
+		}
+		/* If buttons with icons will fit, bring them back*/
+		else if (tasklist_width / buttons > icon_size * 2)
+		{
+			GtkWidget *widget;
+			GList* children = gtk_container_get_children (GTK_CONTAINER(outer_box));
+			while (children != NULL)
+			{
+				button = GTK_WIDGET (children->data);
+				box = gtk_bin_get_child (GTK_BIN (button));
+				GList* contents = gtk_container_get_children (GTK_CONTAINER(box));
+				while (contents != NULL)
+				{
+					widget = GTK_WIDGET (contents->data);
+					if (GTK_IS_LABEL (widget))
+						gtk_widget_hide (widget);
+
+					if (GTK_IS_IMAGE (widget))
+						gtk_widget_show (widget);
+
+					contents = contents->next;
+					gtk_widget_show (box);
+					gtk_widget_show (button);
+				}
+
+				children = children->next;
+				gtk_widget_set_size_request (button, tasklist_width / buttons, -1);
+			}
+		}
+		/* If we still cannot fit labels or icons, just fill the available space*/
+		else
+		{
+			GList* children = gtk_container_get_children (GTK_CONTAINER(outer_box));
+			while (children != NULL)
+			{
+				button = GTK_WIDGET (children->data);
+					gtk_widget_set_size_request (button, tasklist_width / buttons, -1);
```

Comment:
indents, line 461 can start same as line above
```suggestion
				button = GTK_WIDGET (children->data);
				gtk_widget_set_size_request (button, tasklist_width / buttons, -1);
```

Reply:
Found and fixed